storage/src/vespa/storage/distributor/messagetracker.cpp

```diff
@@ -19,10 +20,11 @@ MessageTracker::~MessageTracker() = default;
 void
 MessageTracker::flushQueue(MessageSender& sender)
 {
-    for (uint32_t i = 0; i < _commandQueue.size(); i++) {
-        _commandQueue[i]._msg->setAddress(api::StorageMessageAddress::create(_cluster_ctx.cluster_name_ptr(), lib::NodeType::STORAGE, _commandQueue[i]._target));
-        _sentMessages[_commandQueue[i]._msg->getMsgId()] = _commandQueue[i]._target;
-        sender.sendCommand(_commandQueue[i]._msg);
+    _sentMessages.resize(_commandQueue.size());
```

Comment:
In the case `flushQueue()` is called multiple times per tracker (doesn't seem to currently be the case), this might be more optimal as `_sentMessages.resize(_sentMessages.size() + _commandQueue.size())` to avoid explicitly sizing the table down before implicitly sizing it up again upon reinsertion.

I've got a small branch going on the side, I can sneak this change in there.

Reply:
Thanks, I be ready to review.